src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.js

```diff
@@ -2,25 +2,70 @@ import pathOr from 'ramda/src/pathOr';
 import pathEq from 'ramda/src/pathEq';
 import assocPath from 'ramda/src/assocPath';
 import insert from 'ramda/src/insert';
+import slice from 'ramda/src/slice';
 import findNthIndex from '#lib/utilities/findNthIndex';
 
-const getBlocks = pathOr([], ['content', 'model', 'blocks']);
-const setBlocks = assocPath(['content', 'model', 'blocks']);
+const getBlocks = pathOr([], ['content', 'blocks']);
+const setBlocks = assocPath(['content', 'blocks']);
 const podcastPromoBlock = {
   type: 'podcastPromo',
   model: {},
 };
 
-const getPromoPosition = pageData =>
-  findNthIndex(7, pathEq(['type'], 'text'), getBlocks(pageData));
+const withListException = (blocks, targetIndex) => {
+  if (!pathEq([targetIndex + 2, 'type'], 'list', blocks)) return targetIndex;
+  if (!pathEq([targetIndex + 3, 'type'], 'paragraph', blocks)) return -1;
+  if (!pathEq([targetIndex + 4, 'type'], 'paragraph', blocks)) return -1;
```

Comment:
Are these 2 checks needed? Just reading up on Ramda so just working out the syntax.

Reply:
We need some checks here - there might be a better way of writing them though!

This function is dealing with the "if the block 2 positions after the desired podcast promo index is a list, we can't put the promo in our initial spot" scenario.

If index + 2 is NOT a list, we just continue to put the promo where we wanted (line 16)

If it is a list, we can't put the promo at targetIndex.  Instead, we need to check if the two blocks after the list are both paragraphs.  If they're not both paragraphs, we don't put the promo anywhere (lines 17 and 18)

If index + 2 is a list, index + 3 is a paragraph, index + 4 is also a paragraph, we reach line 19.

Line 19 puts the promo after the list (ie, at index + 3) where it will then float to the left of the two paragraphs after the list.

Reply:
Ah excellent, thanks for the explanation! Yea that makes sense 👍🏼

Reply:
Could potentially make these named constants for readability? Something like `isBlockTwoItemsAfterPromoTypeList`
`isBlockThreeItemsAfterPromoTypeText`
`isBlockFourItemsAfterPromoTypeText`

Although they do look a bit janky lol, might not be beneficial.